airbyte-integrations/connectors/source-stripe/source_stripe/partition.py

```diff
@@ -0,0 +1,141 @@
+            if next_page_token:
+                additional_parameters[self._paginator.page_token_option.field_name] = next_page_token["next_page_token"]
+
+            if self._paginator.page_token_option.inject_into == RequestOptionType.request_parameter:
+                request_params = request_params | additional_parameters
+            elif self._paginator.page_token_option.inject_into == RequestOptionType.header:
+                request_headers = request_headers | additional_parameters
+            elif self._paginator.page_token_option.inject_into == RequestOptionType.body_data:
+                request_body_data = request_body_data | additional_parameters
+            elif self._paginator.page_token_option.inject_into == RequestOptionType.body_json:
+                request_body_json = request_body_json | additional_parameters
+
+            response = self._requester.send_request(
+                path=path,
+                request_headers=request_headers,
+                request_params=request_params,
+                request_body_data=request_body_data,
+                request_body_json=request_body_json,
+                next_page_token=next_page_token,
+                log_formatter=log_formatter,
+            )
+            if not response:
+                pagination_complete = True
+            else:
+                records = list(self._record_selector.select_records(response, {}))
+                yield from records
+
+                next_page_token = self._paginator.next_page_token(response, records)
+                if not next_page_token:
+                    pagination_complete = True
+
+
+class SourcePartitionGenerator(PartitionGenerator):
+    def __init__(
+        self,
+        stream: Stream,
+        requester: PaginatedRequester,
+        request_headers: Optional[Mapping[str, Any]] = None,
+        request_params: Optional[Mapping[str, Any]] = None,
+        request_body_data: Optional[Union[Mapping[str, Any], str]] = None,
+        request_body_json: Optional[Mapping[str, Any]] = None,
+    ):
+        self._stream = stream
+        self._requester = requester
+        self._request_headers = request_headers
+        self._request_params = request_params
+        self._request_body_data = request_body_data
+        self._request_body_json = request_body_json
+
+    def generate(self, sync_mode: SyncMode) -> Iterable[Partition]:
+        for s in self._stream.stream_slices(sync_mode=sync_mode):
+            yield SourcePartition(
+                copy.deepcopy(s),
+                self._stream,
+                self._requester,
+                request_headers=self._request_headers,
+                request_params=self._request_params,
+                request_body_data=self._request_body_data,
+                request_body_json=self._request_body_json,
+            )
+
+
+class SourcePartition(Partition):
+    def __init__(
+        self,
+        _slice: Mapping[str, Any],
+        stream: Stream,
```

Comment:
It would be preferable for the`SourcePartition` to not depend on the `Stream` class so we can eventually stop using it entirely. Let's directly pass the transformer instead